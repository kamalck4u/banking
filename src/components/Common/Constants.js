export const ACCOUNT_NAME = 'Account Name';
export const ACCOUNT_TYPE = 'Account Type';
export const ACCOUNT_NUMBER = 'Account Number';
export const CURRENCY = 'Currency';
export const ACCOUNT_BALANCE = 'Balance';
export const REFERENCE_NUMBER = 'Transaction#';
export const DATE = 'Date';
export const ACCOUNT_FROM = 'From';
export const DESCRIPTION = 'Description';
export const STATUS = 'Status';
export const AMOUNT = 'Amount';
export const WELCOME_TEXT = 'WELCOME CONTOUR';
export const TAG_LINE = 'Building the Future of Trade Finance';
export const ERROR_HEADER = 'Error';
export const TRANSACTIONS_DETAILS = 'Transaction Details';
export const TRANSACTIONS_HISTORY = 'Transaction History';
export const ERROR_DESCRIPTION = 'Some thing went wrong while loading data, Please try reload or login  again !!!';
export const ERROR_ADDITONAL_INFO = 'if the issue persists, Please reach out to Customer Care on +65 9999 8888 or Relationship manager';
export const LOADER_TEXT_ACCOUNTS = 'Your portfolio is being loaded, please wait...';
export const LOADER_TEXT_TRANSACTIONS = 'Your transactions are being loaded, please wait...';

