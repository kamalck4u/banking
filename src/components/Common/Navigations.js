import React from 'react';
import { useHistory } from 'react-router-dom';

const Navigations = ({ home, back }) => {
	const history = useHistory();
	return (
		<div style={{ display: 'inline' }}>
			{
				home && <i className="fa fa-home fa-2x nav-icon" aria-hidden="true" onClick={() => history.push(home)} />
			}
			{
				back && <i className="fa fa-chevron-left fa-2x nav-icon" aria-hidden="true" onClick={() => history.goBack()} />
			}

		</div>
	);
};

export default Navigations;

