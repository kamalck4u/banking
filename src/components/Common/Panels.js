import React from 'react';
import { ERROR_HEADER, ERROR_DESCRIPTION, ERROR_ADDITONAL_INFO } from './Constants';

export const HeaderPanel = ({ header, content, description }) => {
	return (
		<div className="jumbotron panel-header">
			<h1 className="display-4">{header}</h1>
			<p className="lead">{content}</p>
			<hr className="my-4" />
			<p>{description} </p>
		</div>
	);
};


export const GenericErrorPanel = () => {
	return (
		<div className="card mb-3">
			<div className="card-body" style={{ textAlign: 'center' }}>
				<h5 className="card-title">
					<i className="fa fa-exclamation-triangle fa-lg" style={{ color: 'red' }} aria-hidden="true">
					</i> {ERROR_HEADER}
				</h5>
				<p className="card-text">{ERROR_DESCRIPTION}</p>
				<p className="card-text"><small className="text-muted">{ERROR_ADDITONAL_INFO}</small></p>
			</div>
		</div>
	);
};


export const Loader = ({ loaderText }) => {
	return (
		<div className="card">
			<div className="card-header">
				<div className="card-body">
					<h3 style={{ textAlign: 'center' }}>{loaderText}</h3>
					<div className="d-flex justify-content-center">
						<div className="spinner-border" role="status">
						</div>
					</div>
				</div>
			</div>
		</div>
	);
};