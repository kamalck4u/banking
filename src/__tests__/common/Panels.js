import React from 'react';
import renderer from 'react-test-renderer';
import { HeaderPanel, GenericErrorPanel, Loader } from '../../components/Common/Panels';

describe('Panels', () => {
	it('snapshot renders HeaderPanel Component', () => {
		const HeaderPanelComponent = renderer.create(<HeaderPanel />);
		expect(HeaderPanelComponent.toJSON()).toMatchSnapshot();
	});
	it('snapshot renders GenericErrorPanel Component', () => {
		const GenericErrorPanelComponent = renderer.create(<GenericErrorPanel />);
		expect(GenericErrorPanelComponent.toJSON()).toMatchSnapshot();
	});
	it('snapshot renders LoaderPanelComponent', () => {
		const LoaderPanelComponent = renderer.create(<Loader />);
		expect(LoaderPanelComponent.toJSON()).toMatchSnapshot();
	});
});