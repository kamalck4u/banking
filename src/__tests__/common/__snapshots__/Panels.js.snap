// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panels snapshot renders GenericErrorPanel Component 1`] = `
<div
  className="card mb-3"
>
  <div
    className="card-body"
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <h5
      className="card-title"
    >
      <i
        aria-hidden="true"
        className="fa fa-exclamation-triangle fa-lg"
        style={
          Object {
            "color": "red",
          }
        }
      />
       
      Error
    </h5>
    <p
      className="card-text"
    >
      Some thing went wrong while loading data, Please try reload or login  again !!!
    </p>
    <p
      className="card-text"
    >
      <small
        className="text-muted"
      >
        if the issue persists, Please reach out to Customer Care on +65 9999 8888 or Relationship manager
      </small>
    </p>
  </div>
</div>
`;

exports[`Panels snapshot renders HeaderPanel Component 1`] = `
<div
  className="jumbotron panel-header"
>
  <h1
    className="display-4"
  />
  <p
    className="lead"
  />
  <hr
    className="my-4"
  />
  <p>
     
  </p>
</div>
`;

exports[`Panels snapshot renders LoaderPanelComponent 1`] = `
<div
  className="card"
>
  <div
    className="card-header"
  >
    <div
      className="card-body"
    >
      <h3
        style={
          Object {
            "textAlign": "center",
          }
        }
      />
      <div
        className="d-flex justify-content-center"
      >
        <div
          className="spinner-border"
          role="status"
        />
      </div>
    </div>
  </div>
</div>
`;
